extensions/skins/conversation_v1/Conversation.js

```diff
@@ -643,6 +646,25 @@ oppia.directive('progressDots', [function() {
           $scope.changeActiveDot($scope.currentDotIndex + 1);
         }
       };
+
+      /* This is needed to watch any change for both number of dots and currentDotIndex. */
+      $scope.$watchGroup(['dots', 'currentDotIndex'], function(oldValue, newValue) {
+        if(newValue !== oldValue) {
+          if($scope.dots.length > $scope.MAX_DOTS) {
+            if($scope.currentDotIndex < $scope.dots.length - $scope.MAX_DOTS) {
+              $scope.leftmostVisibleDotIndex = $scope.currentDotIndex;
```

Comment:
Actually I'm not sure this is right. Suppose there are 20 cards (0-indexed from 0 to 19). The user goes to card 2 then clicks the left arrow then the left arrow again (so, now on card 0). The user now clicks card 8 (say). The dots showing should still be 0 to 17.

How about instead having changeActiveDot(), decrementCurrentDotIndex(), incrementCurrentDotIndex() make the appropriate adjustments to leftmostVisibleDotIndex and rightmostVisibleDotIndex?